src/json_to_byte_array.py

```diff
@@ -0,0 +1,41 @@
+import getopt
+import os
+import sys
+
+
+def main(argv):
+    input_directory = ''
+    output_directory = ''
+    try:
+        opts, args = getopt.getopt(argv,"hi:o:",["ifile=","ofile="])
+    except getopt.GetoptError:
+        print('ERROR: Bad arguments for json_to_byte_array.py: -i <input_directory> -o <output_directory>')
+        sys.exit(2)
+
+    for opt, arg in opts:
+        if opt == '-h':
+            print('json_to_byte_array.py -i <input_directory> -o <output_directory>')
+            sys.exit()
+        elif opt in ("-i", "--ifile"):
+            input_directory = arg
+        elif opt in ("-o", "--ofile"):
+            output_directory = arg
+
+    for file in os.listdir(input_directory):
+        filename, extension = os.path.splitext(file)
+        if extension == ".json":
+            source_path = os.path.join(input_directory, file)
+            output_file = os.path.join(output_directory, filename + '.cpp')
+            with open(source_path, 'r') as f:
+                module_content = f.read()
+            array = bytearray(module_content.encode('utf-8'))
+
+            with open(output_file, 'wb') as f:
+                print('Created module', output_file, 'from', filename)
+                f.write(array)
```

Comment:
I don't think this is correct. You are essentially
```
read(file.json) -> convert it to bytearray -> write(file.cpp, bytearray)
```
which is in the end operation that just copies the contents from the json files to cpp files. Bytearray is just a different representation of the same data. What we want is to **preprocess** the json data to a valid C++ language constructs, that can be then used from the C++ yaramod without even accessing the filesystem.

But in your case, if you open generated .cpp files, they are just plain json objects.

However, we have some limitations to keep yaramod portable:
- you must use a construct that allows for "literals" with arbitrary size, as we discussed windows compilers can impose a limitation on a length of a string literal

For example you should be able to use arrays, choosing a proper type is up to you:
```cpp
char MATH_MODULE[] = ['{', '\n', '\t', 'k', 'i', 'n', 'd', ':', ...]
```
then in the `module-pool` you would need to convert it to some type that is compatible with your json library. Maybe using `reinterpret_cast` or some other `*_cast` or doing it by "manually" with C++ function for example. 

After the conversion you should have access to the original `file.json`. I know this might sound complicated but we are doing this to ensure cross platform compatibility and users of our library won't have to do anything to adapt with those changes.

Reply:
I have changed the process similarly as suggested:
The `json_to_array.py` python script goes through `/modules` directory and processes all modules that are present there as `.json` (public modules). Each file is split per char and the sequence of chars is written in a new file `module_<name>_generated.h` that contains the following class:
```
class Gmodule_<name> : public ModuleContent
{
public:
	Gmodule_<name>() : ModuleContent("pe", {'{', '\n', ' ', ' ', ' ', ' ', '\"', 'k', 'i', 'n', 'd', '\"', ':', ' ', '\"', 's', 't', 'r', 'u', 'c', 't',...})
	{
	}
};
```
where `<name>` is shortened name of the input `.json` file and `ModuleContent` is a simple base class prepared to store the sequence. These files are generated in the `/include/yaramod/types/modules/generated` directory together with one additional file `modules_list.h` which includes all generated `.h` modules and creates a vector of `ModuleContent` containing an instance of each of the created modules.